### PR TITLE
ci: try rust-cache

### PR DESCRIPTION
**Pull request by bartlomieju**
Prompted by https://matklad.github.io/2021/09/04/fast-rust-builds.html

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -246,53 +246,9 @@ jobs:
...
-            g-cargo-target-${{ matrix.os }}-${{ matrix.profile }}-
-
-      # Don't save cache after building PRs or branches other than 'main'.
-      - name: Skip save cache (PR)
-        run: echo "CACHE_SKIP_SAVE=true" >> $GITHUB_ENV
-        shell: bash
-        if: github.ref != 'refs/heads/main'
-
+      - name: Rust cache
+        uses: Swatinem/rust-cache@v1
```
Comment (dsherret) [inline .github/workflows/ci.yml:250]: I'm not sure why, but when it goes to post the cache it says:

```
Post job cleanup.
Warning:  The "path" argument must be of type string. Received type undefined
```

Comment (bartlomieju) [reply .github/workflows/ci.yml:250]: Not sure either, but it seems to work. I'll push a few more empty commit here to see how it works.

**Review by dsherret — APPROVED** [review]

**Comment by bartlomieju** [comment]
@piscisaureus please take a look

**Comment by AaronO** [comment]
@bartlomieju What's the expected delta ?

**Comment by bartlomieju** [comment]
> @bartlomieju What's the expected delta ?

I don't have concrete expectation in mind; if the time is similar to current ones I'd say it's already a small win, as the caching is one-line with this solution. We can iterate further using suggestions from the blog post. I already found a few crates that take a lot of time to compile:

132138315-268210ab-8846-4c81-80fe-7c4d35b05db4

**Comment by piscisaureus** [comment]
Let's discuss on our Thursday call.

**Comment by bartlomieju** [comment]
Discussed with @piscisaureus, this PR doesn't help.

**Pull request closed without merging**